### PR TITLE
Add type propagation implementation introduced in TF 2.8

### PR DESCRIPTION
**Pull request by klecki**
## Description:
The change in TF dataset is introduced in the two commits:
https://github.com/tensorflow/tensorflow/commit/0e255e18310fd048dbf2bae9b39c1f8858c2a78d
https://github.com/tensorflow/tensorflow/commit/c365301a0d2dd47f1048e1eb81c863d52363841f

Implement it conditionally starting from TF 2.8,
based on the ZipDataset that also handles multiple inputs.

This additionally fixes GPU-placed datasets for TF 2.9.

Signed-off-by: Krzysztof Lecki <klecki@nvidia.com>

### Affected modules and functionalities:
TF Dataset

### Key points relevant for the review:
N/A

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by klecki** [comment]
!build

**Review by JanuszL — APPROVED** [review]

File: dali_tf_plugin/dali_dataset_op.cc [hunk 1]
```diff
@@ -41,6 +41,7 @@
 #include "tensorflow/core/framework/types.h"
 
 #include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/framework/full_type_util.h"
```
Comment (JanuszL) [inline dali_tf_plugin/dali_dataset_op.cc:44]:
```suggestion,
#if TF_MAJOR_VERSION > 2 || (TF_MAJOR_VERSION == 2 && TF_MINOR_VERSION >= 8)
#include "tensorflow/core/framework/full_type_util.h"
#endif
```

Comment (JanuszL) [inline dali_tf_plugin/dali_dataset_op.cc:44]:
```suggestion
```suggestion,
#if TF_MAJOR_VERSION > 2 || (TF_MAJOR_VERSION == 2 && TF_MINOR_VERSION >= 8)
#include "tensorflow/core/framework/full_type_util.h"
#endif
```
```

Comment (klecki) [reply dali_tf_plugin/dali_dataset_op.cc:44]: Right :man_facepalming:

**Comment by klecki** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

**Pull request merged**